sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java

```diff
@@ -322,6 +323,26 @@
     }
   }
 
+  public static <T> void coderConsumesBytesEqualToBytesEncoded(Coder<T> coder, T value)
+      throws IOException {
+
+    byte[] bytes = encode(coder, Coder.Context.NESTED, value);
+
+    @SuppressWarnings("unchecked")
+    Coder<T> deserializedCoder = Serializer.deserialize(coder.asCloudObject(), Coder.class);
+
+    byte[] extendedBytes = new byte[bytes.length + Ints.BYTES];
+    System.arraycopy(bytes, 0, extendedBytes, 0, bytes.length);
+    System.arraycopy(Ints.toByteArray(1), 0, extendedBytes, bytes.length, Ints.BYTES);
+
+    ByteArrayInputStream is = new ByteArrayInputStream(extendedBytes);
+    int startPos = is.available();
```

Comment:
available doesn't need to tell the truth, better to rely on a CountingInputStream to record how many bytes were actually read.

https://google.github.io/guava/releases/19.0/api/docs/com/google/common/io/CountingInputStream.html